crates/consensus/src/transaction/envelope.rs

```diff
@@ -178,56 +182,63 @@ pub enum TxEnvelope {
     ///
     /// 2 - The transaction with a sidecar, which is the form used to
     /// send transactions to the network.
-    Eip4844(Signed<TxEip4844Variant>),
+    Eip4844(Signed<Eip4844>),
     /// A [`TxEip7702`] tagged with type 4.
     Eip7702(Signed<TxEip7702>),
 }
-
-impl From<Signed<TxLegacy>> for TxEnvelope {
+pub type TxEnvelope = EthereumTxEnvelope<TxEip4844Variant>;
```

Comment:
can we move this top of file

and add some docs, that this type supports all EIP4844 variants